### PR TITLE
Fix Issue: group node in fbx being exported as bone node

### PR DESCRIPTION
**Pull request by thomasbiang**
Issue #2537: using Assimp, simply import an fbx file then export it without any modification, the output fbx file could be different from the original one: group node (being a parent of any bone node) in original fbx is exported as bone node in the output fbx. 

Solution: for any node, if it is not a bone node in the scene, and if its transformation is identity matrix (means: does not contribute to the world transformation computation for its children), these nodes should not be exported as bones.  

Code changes: minimal. Use a set to record all bone names in scene. When exporting, check each node, if its name is not among these bone names and its transformation is identity matrix; then it will not be exported as bone node.

### CODE REVIEW

File: code/FBX/FBXExporter.cpp [hunk 1]
```diff
@@ -1617,6 +1617,17 @@ void FBXExporter::WriteObjects ()
     // at the same time we can build a list of all the skeleton nodes,
     // which will be used later to mark them as type "limbNode".
     std::unordered_set<const aiNode*> limbnodes;
+    
+    //actual bone nodes in fbx, without parenting-up
+    std::set<std::string> setAllBoneNamesInScene;
```
Comment (turol) [inline code/FBX/FBXExporter.cpp]: Prefer `std::unordered_set` where ordering is not required.

File: code/FBX/FBXExporter.cpp [hunk 2]
```diff
@@ -1617,6 +1617,17 @@ void FBXExporter::WriteObjects ()
...
+    for(int m = 0; m < mScene->mNumMeshes; ++ m)
```
Comment (turol) [inline code/FBX/FBXExporter.cpp]: Signed vs unsigned compare warning.

File: code/FBX/FBXExporter.cpp [hunk 3]
```diff
@@ -1617,6 +1617,17 @@ void FBXExporter::WriteObjects ()
...
+    {
+        aiMesh* pMesh = mScene->mMeshes[m];
+        for(int b = 0; b < pMesh->mNumBones; ++ b)
```
Comment (turol) [inline code/FBX/FBXExporter.cpp]: Same, signed vs unsigned.

**Comment by thomasbiang** [comment]
Hi, 

I just updated code changes according to the comments. 

Thank you for reviewing

**Comment by kimkulling** [comment]
Finally merged!

**Pull request merged**